### PR TITLE
Fix crash when processing PostgreSQL packets in weird order

### PR DESCRIPTION
**Pull request by iamnotacake**
* Clear `pendingParse` after receiving `ParseComplete`
* Clear `pendingBind` (sensitive data inside) after receiving `BindComplete`
* Clear `pendingBind` (set to `nil`) after receiving `ReadyForQuery`

Also, tiny fix in integration tests so `DEBUG_LOG=False` env var will really make AcraServer logs less verbose.

## Checklist

- [ ] ~Change is covered by automated tests~ The bug that was fixed here only happens with Rust [`postgres`](https://lib.rs/crates/postgres) driver, manually ensured it does not reproduce any more
- [x] The [coding guidelines] are followed
- [x] ~Public API has proper documentation in the [Acra documentation] site or has PR on [documentation repository] 
  with new changes~
- [x] ~CHANGELOG.md is updated (in case of notable or breaking changes)~
- [x] CHANGELOG_DEV.md is updated
- [x] ~Benchmark results are attached (if applicable)~
- [x] ~[Example projects and code samples] are up-to-date (in case of API changes)~

[coding guidelines]: https://golang.org/doc/effective_go
[Example projects and code samples]: https://github.com/cossacklabs/acra-engineering-demo
[Acra documentation]: https://docs.cossacklabs.com/
[documentation repository]: https://github.com/cossacklabs/product-docs

### CODE REVIEW

**Comment by vixentael** [comment]
shouldn't we use mutex anywhere?

**Comment by Lagovas** [comment]
> shouldn't we use mutex anywhere?
mutexes should be used to synchronize access to shared objects. Here is data related to 2 connections (app <> acra, acra <> db), and handled by 2 goroutines. If 2 goroutines change/update same data - then we should. If only one of them touches data at same time - then we shouldn't. For example if we touch data related to client's side packets in goroutine that work with client side connection, then we can omit mutexes because of opposite goroutine will not touch same data/buffers/processes.

File: decryptor/postgresql/protocol.go [hunk 1]
```diff
@@ -151,19 +154,34 @@ func (p *PgProtocolState) HandleDatabasePacket(packet *PacketHandler) error {
...
+		}
+
+		// Forget pending bind, but only if we got BindComplete before
+		if p.gotBindComplete {
+			p.forgetPendingBind()
+			p.gotBindComplete = false
+		}
+
+		p.forgetPendingExecute()
+		p.forgetPendingQuery()
```
Comment (Lagovas) [inline decryptor/postgresql/protocol.go:184]: should we forget query after ParseComplete (actually packet that stores query)? Or any next packets depends on that (Bind/Execute)?

File: tests/test.py [hunk 2]
```diff
@@ -1213,6 +1213,7 @@ def get_vault_cli_args(self, mount_path=None, secret_path=None):
 
 class BaseTestCase(PrometheusMixin, unittest.TestCase):
     DEBUG_LOG = os.environ.get('DEBUG_LOG', True)
+    if DEBUG_LOG == 'False' or DEBUG_LOG == '0': DEBUG_LOG = False
```
[the remark this answers is not in the document]
Comment (iamnotacake) [reply tests/test.py]: Done

**Review by Lagovas — APPROVED** [review]

**Pull request merged**